mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/mapping/building/spi/PojoMetadataContributor.java

```diff
@@ -15,4 +15,19 @@
 
 	void contributeMapping(MAC collector);
 
+	/**
+	 * Using this method the contribution to the mapping is applied before any other contribution
+	 * provided with {@link #contributeMapping(PojoMappingCollector)}.
+	 * <p>
+	 * Used only by contributions that must be provided before other contributions on the same collector.
+	 * For instance when we need to provide <b>a mapping default value</b> that could be overridden by an another contributor.
+	 * <p>
+	 * Use {@link #contributeMapping(PojoMappingCollector)} whenever a contribution could be applied at any time.
+	 *
+	 * @param collector the target collector for the contribution.
+	 */
+	default void contributePreMapping(MAC collector) {
```

Comment:
I don't understand why you added this method. Contributing metadata before the mapping is exactly the point of `contributeAdditionalMetadata`. Let's talk about it, I think there's a misunderstanding.